Counter/src/app/pokemon-thumbnail/pokemon-thumbnail.component.ts

```diff
@@ -0,0 +1,43 @@
+import { Component, OnInit, Input } from '@angular/core';
+import {HttpClient} from '@angular/common/http';
+import { Pokemon } from '../pokemon';
+import { Data } from '@angular/router';
+
+@Component({
+  selector: 'app-pokemon-thumbnail',
+  templateUrl: './pokemon-thumbnail.component.html',
+  styleUrls: ['./pokemon-thumbnail.component.css']
+})
+export class PokemonThumbnailComponent implements OnInit {
+
+  @Input()
+  pokemon: Pokemon = new Pokemon;
+  image!: String;
+
+  constructor(private http:HttpClient) {
+  }
+
+  ngOnInit(): void {
+    this.getImageUrl();
+  }
+
+  getImageUrl() {
+    var lowerName = new String(this.pokemon.name);
+    lowerName  = lowerName[0].toLowerCase() + lowerName.slice(1);
+    this.http.get<JSON>("https://pokeapi.co/api/v2/pokemon/" + lowerName).subscribe(data=> {
```

Comment:
instead of using JSON you should use a type that you create. One that has sprites, other, home, front_shiny on it. Then you can use pipe to map down to just the image you care about. Typically your httpclient access is hidden behind a service so you can easily test this component. You should try putting this in a pokemon service so you can hide the http access.

Take a look at tour of heroes where they talk about type casting a response into your own model. https://angular.io/guide/http#requesting-a-typed-response

Additionally, all subscribes must be unsubscribed otherwise it will listen to that observable forever. In this case you may be better off using the async pipe method and not subscribing so that angular manages the lifecycle. https://angular.io/tutorial/toh-pt6#asyncpipe